client/blocks/sharing-preview-pane/index.jsx

```diff
@@ -0,0 +1,133 @@
+/**
+ * External dependencies
+ */
+import React, { PureComponent, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get, find } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getPostImage } from './utils';
+import FacebookSharePreview from 'components/share/facebook-share-preview';
+import TwitterSharePreview from 'components/share/twitter-share-preview';
+import VerticalMenu from 'components/vertical-menu';
+import { SocialItem } from 'components/vertical-menu/items';
+import { getSitePost } from 'state/posts/selectors';
+import { getSeoTitle } from 'state/sites/selectors';
+import { getSite } from 'state/sites/selectors';
+import { getSiteUserConnections } from 'state/sharing/publicize/selectors';
+import { getCurrentUserId } from 'state/current-user/selectors';
+
+class SharingPreviewPane extends PureComponent {
+
+	static propTypes = {
+		siteId: PropTypes.number,
+		postId: PropTypes.number,
+		services: PropTypes.array,
+		message: PropTypes.string,
+		// connected properties
+		site: PropTypes.object,
+		post: PropTypes.object,
+		seoTitle: PropTypes.string,
+	};
+
+	static defaultProps = {
+		services: [ 'facebook', 'twitter' ]
+	};
+
+	state = {
+		selectedService: 'facebook'
+	};
+
+	selectPreview = ( selectedService ) => {
+		this.setState( { selectedService } );
+	};
+
+	renderPreview() {
+		const { post, message, connections } = this.props;
+		const { selectedService } = this.state;
+		const connection = find( connections, { service: selectedService } );
+		if ( ! connection ) {
+			return null;
+		}
+
+		const articleUrl = get( post, 'URL', '' );
+		const imageUrl = getPostImage( post );
+		const {
+			external_name: externalName,
+			external_profile_url: externalProfileURL,
+			external_profile_picture: externalProfilePicture,
+			external_display: externalDisplay,
+		} = connection || {};
```

Comment:
This or `|| {}`  part seems to be unneeded since if `connection` isn't valid the `renderPreview` method will return `null`.